src/crab/wto_nesting.hpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Prevail Verifier contributors.
+// SPDX-License-Identifier: MIT
+#pragma once
+
+// Bournacle, "Efficient chaotic iteration strategies with widenings", 1993
+// uses the notation w(c) to refer to the set of heads of the nested components
+// containing a vertex c.  This class holds such a set of heads.  The table
+// mapping c to w(c) is stored outside the class, in wto_t._nesting.
+class wto_nesting_t final {
+    // To optimize insertion performance, the list of heads is stored in reverse
+    // order, i.e., from innermost to outermost cycle.
+    std::vector<label_t> _heads;
+
+  public:
+    wto_nesting_t(std::vector<label_t>&& heads) : _heads(heads) {}
```

Comment:
If I am not mistaken, we still need `std::move(heads)` here. Even though the type of `heads` is an rvalue reference, named references are always lvalue reference as a safety precaution, so it needs the cast again, otherwise it will be copied.

Quoting https://en.cppreference.com/w/cpp/language/value_category
> [T]he name of a variable, a function, a template parameter object (since C++20), or a data member, regardless of type, such as std::cin or std::endl. Even if the variable's type is rvalue reference, the expression consisting of its name is an lvalue expression;

I believe that function parameters are considered variables in this definition.

```suggestion
    wto_nesting_t(std::vector<label_t>&& heads) : _heads(std::move(heads)) {}
```